db/dataset_eval.py

```diff
@@ -62,14 +63,20 @@ def evaluate_dataset(dataset_id, normalize, eval_location, filter_type=None):
 
     Returns:
         ID of the newly created evaluation job.
+        Dictionary of class name and corresponding list of ignored recordings
     """
     with db.engine.begin() as connection:
         if _job_exists(connection, dataset_id):
             raise JobExistsException
 
+        dataset = db.dataset.get(dataset_id)
+        # Filter dataset
+        dataset, ignore_recordings = filter_dataset_contents(dataset)
         # Validate dataset contents
-        validate_dataset_contents(db.dataset.get(dataset_id))
-        return _create_job(connection, dataset_id, normalize, eval_location, filter_type)
+        validate_dataset_structure(dataset)
```

Comment:
(we talked about this on irc)
this will raise an exception if the dataset structure isn't complete - after doing a filter, some class may have less than 2 recordings. This means that you don't get to send the flash message about recordings being removed, and the user only sees a message about the dataset being invalid, which is a bit confusing.